src/data/queries/mediaImage.ts

```diff
@@ -0,0 +1,50 @@
+import {
+  QueryData,
+  QueryFormatter,
+  QueryOpts,
+  QueryParams,
+} from 'next-drupal-query'
+import { queries } from '.'
+import { MediaImage } from '@/types/dataTypes/drupal/media'
+import { drupalClient } from '@/lib/utils/drupalClient'
+import { MediaImageType } from '@/types/index'
+
+// Define query params for queryData.
+export const params: QueryParams<null> = () => {
+  return queries.getParams().addPageLimit(12).addInclude(['image'])
+}
+// Define the option types for the data loader.
+type DataOpts = QueryOpts<null>
+
+// Implement the data loader.
+export const data: QueryData<
+  DataOpts,
+  MediaImage
+> = async (): Promise<MediaImage> => {
+  const entities = await drupalClient.getResourceCollection<MediaImage>(
+    'media--image',
+    {
+      params: params().getQueryObject(),
+    }
+  )
+  return entities
+}
+
+export const formatter: QueryFormatter<MediaImage, MediaImageType[]> = (
+  entities: MediaImage
+) => {
+  if (!entities) return null
+
+  const castArray = (val) => (Array.isArray(val) ? val : [val])
```

Comment:
this formatter is the only part that jumps out at me still, feels a little weird to need to cast to array and then potentially null/undefined values for all of the `.image?` checks... but i haven't run it locally yet. if this is working i dont want it to be a blocker

Reply:
Yeah lets talk about this more during the next build sync. The slug route pages return data as an object vs data being returned as an collection array on no slug rout pages. Working with the new story listings is where the data structure is a little bit odd. 

```
{
0: {
id: 3
  link : {
      href: https://
      meta: {
        linkParams : {
         width: 100
         height: 100
        }
      }
    }
  }
}
```